modules/icgc-argo-workflows/checkinput/resources/usr/bin/rnaaln.py

```diff
@@ -0,0 +1,491 @@
+        """
+        Perform all validations on the given row and insert the read pairing status.
+
+        Args:
+            row (dict): A mapping from column headers (keys) to elements of that row
+                (values).
+
+        """
+        self._validate_analysis_type(row) if row.get(self._analysis_type_col) else ""
+        self._validate_sex(row) if row.get(self._sex_col) else ""
+        self._validate_study_id(row) if row.get(self._study_id_col) else ""
+        self._validate_patient(row) if row.get(self._patient_col) else ""
+        self._validate_sex(row) if row.get(self._sex_col) else ""
+        self._validate_status(row) if row.get(self._status_col) else ""
+        self._validate_sample(row)
+        self._validate_lane(row)
+        self._validate_single_end(row)
+        self._validate_fastq_1(row)
+        self._validate_fastq_2(row)
+        self._validate_library_name(row)
+        self._validate_platform_unit(row)
+        self._validate_platform_col(row) if row.get(self._platform_col) else ""
+        self._validate_sequencing_center_col(row) if row.get(self._sequencing_center_col) else ""
+        self._validate_sequencing_date_col(row) if row.get(self._sequencing_date_col) else ""
+        self._validate_platform_model_col(row) if row.get(self._platform_model_col) else ""
+        self._validate_read_group_count(row)
+        self._validate_experiment(row) if row.get(self._experiment_col) else ""
+        self._validate_analysis_json(row) if row.get(self._analysis_json_col) else ""
+        self._validate_library_strandedness_col(row)
+
+        tmp_dict={
+            "analysis_type" : row[self._analysis_type_col] if row.get(self._analysis_type_col) else "sequencing_experiment",
+            "study_id" : row[self._study_id_col] if row.get(self._study_id_col) else "LOCAL",
+            "patient" : row[self._patient_col] if row.get(self._patient_col) else row[self._sample_col],
+            "sex" : row[self._sex_col] if row.get(self._sex_col) else "NA",
+            "status" : row[self._status_col] if row.get(self._status_col) else "0",
+            "sample" : row[self._sample_col],
+            "lane" : row[self._lane_col],
+            "fastq_1" : row[self._fastq_1_col],
+            "fastq_2" : row[self._fastq_2_col] if row.get(self._fastq_2_col) else "NO_FILE",
+            "single_end" : row[self._single_end_col].lower(),
+            "read_group_count" : row[self._read_group_count_col],
+            "experiment" : row[self._experiment_col] if row.get(self._experiment_col) else "WGS",
+            "analysis_json" : row[self._analysis_json_col] if row.get(self._analysis_json_col) else None,
+            "library_strandedness": row[self._library_strandedness_col] if row.get(self._library_strandedness_col) else None,
+            }
+
+        # Transform 'library_strandedness' based on its value after dictionary creation
+        if tmp_dict['library_strandedness'] == "FIRST_READ_SENSE_STRAND":
```

Comment:
- move into a function similar to other see below `_validate_single_end`
- add error case when library strandedness value is outside of expected
- account for unstranded : https://github.com/alexdobin/STAR/issues/258#issuecomment-296519787